[ASM] Add clarifications to the API Security page (#25302)
DataDog/documentation: content/en/security/application_security/api-inventory/_index.md

## Configuration

> In order to use API Security on your services, **you must have ASM Threats Protection enabled**. The following library versions are compatible with API Security Inventory. [Remote Configuration][1] is required.

Reviewer: ```suggestion
To use API Security on your services, **you must have ASM Threats Protection enabled**. The following library versions are compatible with API Security Inventory. [Remote Configuration][1] is required.
```

|Golang    | v1.59.0  | Requests only |
|Node.js   | v3.51.0, v4.30.0 or v5.6.0 | Requests and responses |

> **Warning** On .NET Core and .NET Fx tracers, you need to set `DD_API_SECURITY_ENABLED=true` in your environement variables for API Security features to work properly.

Reviewer: ```suggestion
**Note**: On .NET Core and .NET Fx tracers, you need to set the environment variable `DD_API_SECURITY_ENABLED=true` for API Security features to work properly.
```
